redbot/core/utils/common_filters.py

```diff
@@ -16,6 +17,8 @@
 
 MASS_MENTION_RE = re.compile(r"(@)(?=everyone|here)")  # This only matches the @ for sanitizing
 
+# negative lookbehind used here to avoid canceling our own filter strategy.
+OTHER_MENTION_RE = re.compile(r"(?<!\\)(<(@?[!&]|#)\d+>)")
```

Comment:
Unfortunately, users and roles are still pinged when there's a backslash in front of the full mention. It only prevents it from being formatted as a mention. Also, the character group for `[!&]` should be optional, not the `!`. Something like this:
```py
r"<(@(?:[!&]?|#)\d+)>"
```
called by:
```py
OTHER_MENTION_RE.sub(r"<\\\1>", to_filter)
```
should do the trick.